Sources/MapboxMaps/Foundation/Extensions/Core/CoordinateBounds.swift

```diff
@@ -5,27 +5,106 @@ import MapboxCoreMaps
+    var west: CLLocationDegrees { __west() }
+
+    /// Returns the northern latitude of the bounds.
+    var north: CLLocationDegrees { __north() }
+
+    /// Returns the eastern longitude of the bounds.
+    var east: CLLocationDegrees { __east() }
+
+    /// The centerpoint of this `CoordinateBounds` calculated by simple interpolation.
+    /// This is a non-geodesic calculation which is not the geographic center.
+    var center: CLLocationCoordinate2D {
+        return __center()
+    }
+
+    /// Returns whether the bounds are empty or not.
+    var isEmpty: Bool { __isEmpty() }
+
+    /// The absolute distance, in degrees, between the north and south boundaries of these bounds.
+    ///
+    /// One degree of latitude is approximately 111 kilometers (69 miles).
+    var latitudeSpan: CLLocationDegrees {
+        __latitudeSpan()
+    }
+
+    /// The absolute distance, in degrees, between the west and east boundaries of these bounds.
+    ///
+    /// The distance represented by a longitude span varies on current latitude.
+    /// At the equator one degree of longitude represents a distance of approximately 111 kilometers (69 miles).
+    /// While at the poles one degree of logitude span is 0 kilometers (0 miles).
+    var longitudeSpan: CLLocationDegrees {
+        __longitudeSpan()
+    }
+
+    /// The northwest coordinate of the bounds.
     var northwest: CLLocationCoordinate2D {
-        return CLLocationCoordinate2D(latitude: northeast.latitude, longitude: southwest.longitude)
+        return __northwest()
     }
 
-    /// The northwest coordinate of an internal `CoordinateBounds` type.
+    /// The southeast coordinate of the bounds.
     var southeast: CLLocationCoordinate2D {
-        return CLLocationCoordinate2D(latitude: southwest.latitude, longitude: northeast.longitude)
+        return __southeast()
     }
-}
 
-internal extension CoordinateBounds {
-    func contains(_ coordinates: [CLLocationCoordinate2D]) -> Bool {
-        let latitudeRange = southwest.latitude...northeast.latitude
-        let longitudeRange = southwest.longitude...northeast.longitude
+    /// Returns a new `CoordinateBounds` that stretches to contain both this and another `CoordinateBounds`.
+    /// - Parameter coordinateBounds: The `CoordinateBounds` to add.
+    /// - Returns: A bounds whose area encompasses the two bounds and the space between them.
+    func union(_ coordinateBounds: CoordinateBounds) -> CoordinateBounds {
+        extend(forArea: coordinateBounds)
+    }
 
-        for coordinate in coordinates {
-            if latitudeRange.contains(coordinate.latitude) || longitudeRange.contains(coordinate.longitude) {
-                return true
-            }
+    /// Extends these bounds to include `point`.
+    /// - Parameter coordinate: The coordinate to be included.
+    /// - Returns: The extended bounds.
+    func include(_ coordinate: CLLocationCoordinate2D) -> CoordinateBounds {
+        extend(forPoint: coordinate)
+    }
```

Comment:
Can you confirm whether these functions return new instances or mutate self?

I'd also like to workshop on these names with you a bit to think through consistency with Android and adherence to the Swift API Design Guidelines.

Reply:
They return new instances.

Yes, the inconsistency between `union` and `include` is puzzling, given that they perform similar things.

Reply:
Removed as a result of our workshop.